editor/code_editor.cpp

```diff
@@ -321,40 +328,52 @@ void FindReplaceBar::_get_search_from(int &r_line, int &r_col) {
 }
 
 void FindReplaceBar::_update_results_count() {
-	if (results_count != -1) {
+	if (!needs_to_count_results && (result_line != -1)) {
+		results_count_to_current += (flags & TextEdit::SEARCH_BACKWARDS) ? -1: 1;
+
+		if (results_count_to_current > results_count)
+			results_count_to_current = results_count_to_current - results_count;
+		else if (results_count_to_current == 0)
+			results_count_to_current = results_count;
+
 		return;
 	}
 
+	needs_to_count_results = false;
+
 	results_count = 0;
+	results_count_to_current = 0;
 
 	String searched = get_search_text();
 	if (searched.is_empty()) {
 		return;
 	}
 
-	String full_text = text_editor->get_text();
+	for (int i=0; i < text_editor->get_line_count(); i++) {
+		String line_text = text_editor->get_line(i);
 
-	int from_pos = 0;
+		int col_pos = 0;
 
-	while (true) {
-		int pos = is_case_sensitive() ? full_text.find(searched, from_pos) : full_text.findn(searched, from_pos);
-		if (pos == -1) {
-			break;
-		}
+		for (;;) {
+			col_pos = is_case_sensitive() ? line_text.find(searched, col_pos) : line_text.findn(searched, col_pos);
 
-		int pos_subsequent = pos + searched.length();
-		if (is_whole_words()) {
-			from_pos = pos + 1; // Making sure we won't hit the same match next time, if we get out via a continue.
-			if (pos > 0 && !(is_symbol(full_text[pos - 1]) || full_text[pos - 1] == '\n')) {
-				continue;
-			}
-			if (pos_subsequent < full_text.length() && !(is_symbol(full_text[pos_subsequent]) || full_text[pos_subsequent] == '\n')) {
-				continue;
+			if (col_pos == -1)
+				break;
+
+			if (is_whole_words()) {
+				if (col_pos > 0 && !is_symbol(line_text[col_pos - 1]))
+					break;
+				if (col_pos + line_text.length() < line_text.length() && !is_symbol(line_text[col_pos + searched.length()]))
+					break;
 			}
-		}
 
-		results_count++;
-		from_pos = pos_subsequent;
+			if ((i <= result_line) && ((i < result_line) || (col_pos <= result_col)))
+				results_count_to_current++;
```

Comment:
```suggestion
			if (i == result_line && col_pos == result_col) {
				results_count_to_current = results_count;
			}
```

Reply:
Ummmmm it is more elegant 🙀

Just needs to be after incrementing `results_count`.
```cpp
results_count++;

if (i == result_line && col_pos == result_col)
    results_count_to_current = results_count;
```